### DISCUSSION TITLE
Error from server (BadRequest): container "pulsar-broker" in pod "pulsar-broker-0" is waiting to start: PodInitializing

### DISCUSSION THREAD

**Discussion by subhranil05** [apache/pulsar#17802] [Q&A]
pulsar image version = 2.9.3
initialize: true  #set to true in values.yaml

Pulsar-broker and proxy pods  are not coming up due to init containers
zookeeper and other pods are up and running
pulsar-broker-0            0/1     Init:1/2    0          5m40s
pulsar-proxy-0             0/1     Init:1/2    0          5m40s

logs :  Defaulted container "pulsar-broker" out of: pulsar-broker, wait-zookeeper-ready (init), wait-bookkeeper-ready (init)
Error from server (BadRequest): container "pulsar-broker" in pod "pulsar-broker-0" is waiting to start: PodInitializing

Please help me by providing solution if there any.  Need real quick!!!!

**Comment by asafm** [comment]
@hangc0276 Maybe you know?

**Comment by lhotari** [comment]
There was on reply on Pulsar Slack to check the logs of the init jobs that initialize the cluster.
It's not very useful to do any analysis without the logs.

It's useful to check Kubernetes events and logs

Tips related to following k8s events in the console:
```
# events for ALL namespaces (fine for a local minikube or Docker Desktop k8s)
kubectl get events --sort-by=.lastTimestamp -A
# tailing all events
kubectl get events --sort-by=.lastTimestamp -A --watch
```

Getting logs, `pulsar-broker-0` in `pulsar` namespace as example:
```
kubectl -n pulsar logs pulsar-broker-0
# getting previous logs, for the previously crashed pod
kubectl -n pulsar logs pulsar-broker-0 -p
```

In this case, I'd check the init job logs (append `-n pulsar` to the command if you have a specific command, check the jobs with `kubectl get jobs` to see if they have different names).
```
kubectl logs job/pulsar-pulsar-init
kubectl logs job/pulsar-pulsar-init -p
kubectl logs job/pulsar-bookie-init
kubectl logs job/pulsar-bookie-init -p
```

You could use a k8s UI such as `k9s` on command line or [OpenLens](https://github.com/MuhammedKalkan/OpenLens)/[Lens](https://k8slens.dev/) for the desktop. That makes things a lot users to begin with.

**Comment by rahuldstk** [comment]
i faced something similar to this, and for me it solved when i deleted the namespace and created again along with the installation steps. 
i had initially installed another version of pulsar in the same namespace, and when i tried this later version, the secrets might have been in conflict.
i deleted the old namespace and created again, and installed pulsar. now all the pods are in running state

**Comment by subhranil05** [comment]
Try it by disabling anti affinity 
`anti_affinity: false`
